mlflow/tracking/_tracking_service/client.py

```diff
@@ -301,12 +294,6 @@ def log_batch(self, run_id, metrics=(), params=(), tags=()):
             return
         if len(params) > 1:
```

Comment:
I left this validation in here so that we don't have a partial commit. Is this ok?

Reply:
`FileStore` performs the same validation.

https://github.com/mlflow/mlflow/blob/0fa849ad75e5733bf76cc14a4455657c5c32f107/mlflow/store/tracking/file_store.py#L885-L889

Reply:
`SqlAlchemyStore` doesn't:

https://github.com/mlflow/mlflow/blob/d748f559b4122bdfba1283ccf3b4847e73f0d602/mlflow/store/tracking/sqlalchemy_store.py#L811-L815

Reply:
`RestStore` performs no validation (is this intentional?):

https://github.com/mlflow/mlflow/blob/e7caa0a1ffd47255f52b2818fad49d34ed332d99/mlflow/store/tracking/rest_store.py#L302-L307

Reply:
Are there any downsides to performing this validation in a backend store?

Reply:
With the SqlAlchemyStore validation not being done, the first of a duplicate key will be written, raising an underlying SQLAlchemy primary key index collision exception. I think the proper thing to do here is to enforce the unique check on SqlAlchemy and remove from the client. I'll push that change soon.

Reply:
(specifically: `sqlite3.IntegrityError: UNIQUE constraint failed: params.key, params.run_uuid`)